### ISSUE TITLE
Improve the `Dealing with DNS leaks` section of the `WireGuard Selective Routing to External VPN Provider` guide

### ISSUE THREAD

**Issue by schnerring** [opnsense/docs#358]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/docs/blob/master/CONTRIBUTING.md
- [x] I have searched the existing issues and I am convinced that mine is new.

**Describe the issue**

The following part of the [Dealing with DNS leaks](https://docs.opnsense.org/manual/how-tos/wireguard-selective-routing.html#dealing-with-dns-leaks) section of the [WireGuard Selective Routing to External VPN Provider](https://docs.opnsense.org/manual/how-tos/wireguard-selective-routing.html) guide should be more clear here IMHO:

> Force the local DNS server to use the tunnel as well. For a local DNS server that is [...] OPNsense itself, configure the DNS server to use the tunnel gateway.

Intuitively, I tried setting the **Outgoing Network Interfaces** Unbound option to the VPN interface. No matter what option I chose, traffic always used the default route. I think it's because this option just doesn't work for WireGuard interfaces. The help text of the option notes:

> Utilize different network interfaces that Unbound will use to send queries to authoritative servers and receive their replies. By default all interfaces are used. **Note that setting explicit outgoing interfaces only works when they are statically configured.**

After tinkering with this forever, my conclusion is that the only way to "configure the DNS server to use the tunnel gateway" is by adding a static route for the remote peer IP to the WAN gateway. After, changing the default gateway to one of the VPN gateways, Unbound traffic would go through the tunnel as expected. But so would the traffic of all other services originating from the firewall itself, like Web GUI debugging tools. I'm not sure how desirable this is but couldn't figure out another way.

**Suggestions**

Can anyone confirm my understanding of the matter is correct?

If so this...

> configure the DNS server to use the tunnel gateway

...is not really possible. Maybe we should rephrase it to something like:

> configure traffic originating from OPNsense itself to use the tunnel gateway by changing the default route / gateway / routing table

**Version affected**

OPNsense 21.7.4-amd64

**Comment by Greelan** [comment]
Possibly you ran into this issue? https://github.com/opnsense/core/issues/5329

**Comment by schnerring** [comment]
I'm on 21.7.4, too. I revert to 21.7.3 and report back. I think 21.7.4 updated the kernel, so I need to rollback the kernel to 21.7.2. Could you confirm the following is correct, please:

```shell
opnsense-update -kr 21.1.2
opnsense-revert 21.1.3 opnsense
```

**Comment by schnerring** [comment]
I can confirm that this is a bug and not a problem of the docs.

**Comment by frozen905** [comment]
Novice user here who is completely confused by the entire bottom of this documentation..  I am just a newbie and it's almost like reading another language.

There are no examples, no diagrams or even elaboration.  I know the trend in the Linux world or whatever is to downvote comments like these but I'm just telling you, as a layman -- the entire DNS leaks section is a disaster, and makes little to no sense to someone like me.

I completed the guide up to this point just fine, but of course now have leaking DNS servers, and don't have a clue what any of this stuff is talking about at the bottom, nor how to resolve my problem.

I'm going to try pasting it all into AI since I am so confused
